txnkv/transaction/txn.go

```diff
@@ -599,13 +599,19 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 
 	if lockCtx.LockOnlyIfExists {
 		if !lockCtx.ReturnValues {
-			return errors.New("LockOnlyIfExists is set for Lock Context, but ReturnValues is not set")
+			return &tikverr.ErrLockOnlyIfExistsNoReturnValue{
+				ForUpdateTs: lockCtx.ForUpdateTS,
+				LockKey:     keysInput[0],
```

Comment:
Need to check `len(keysInput) > 0` first?

Reply:
if len(keysInput) == 0, it prints "{}".

Reply:
if len(keysInput) == 0, would `keysInput[0]` cause `index out of range`?

Reply:
done